core/db/default/spree/states.rb

```diff
@@ -1,12 +1,84 @@
-Spree::Country.where(states_required: true).each do |country|
-  carmen_country = Carmen::Country.named(country.name)
-  next unless carmen_country
+# frozen_string_literal: true
 
-  carmen_country.subregions.each do |subregion|
+def state_level(subregion, country)
+  country.states.where(
+    name: subregion.name,
+    abbr: subregion.code,
+    country_id: country.id
+  ).first_or_create
+end
+
+def province_level(subregion, country)
+  subregion.subregions.each do |province|
+    country.states.where(
+      name: province.name,
+      abbr: province.code,
+      country_id: country.id
+    ).first_or_create
+  end
+end
+
+def us_state_curation(subregion, country)
+  # Produces 50 states, 1 postal distric (Washingdon DC)
+  # and 3 APO's as you would expect to see on any good U.S. states list.
+  case subregion.code
+  when 'UM'
+    nil
+  when 'AS'
+    nil
+  when 'MP'
+    nil
+  when 'VI'
+    nil
+  when 'PR'
+    nil
+  when 'GU'
+    nil
+  else
+    country.states.where(
+      name: subregion.name,
+      abbr: subregion.code,
+      country_id: country.id
```

Comment:
@MatthewKennedy we don't need to pass country_id here as this scope is already set via `country.states`